File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.data = [[int(cell) for cell in row.split(" ")] for row in matrix_string.split("\n")]

    def row(self, index):
        return self.data[index - 1]

    def column(self, index):
        return [row[index - 1] for row in self.data]

File: fsharp/matrix/Matrix.fs
module Matrix

let private parse (matrix: string) =
    matrix.Split("\n")
    |> Array.map (fun s -> s.Split(" "))
    |> array2D
    |> Array2D.map int

// https://stackoverflow.com/a/2369690/1595197
let private flatten (values: 'a[,]) = 
    values
    |> Seq.cast<'a>
    |> Seq.toList

let row index matrix =
    parse matrix
    |> fun m -> m.[index - 1 .. index - 1, *]
    |> flatten

let column index matrix =
    parse matrix
    |> fun m -> m.[*, index - 1 .. index - 1]
    |> flatten
